presto-main/src/main/java/com/facebook/presto/cost/PlanCostEstimate.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.cost;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.lang.Double.isNaN;
+
+public final class PlanCostEstimate
+{
+    public static PlanCostEstimate zero()
+    {
+        return new PlanCostEstimate(0, 0, 0, 0);
+    }
+
+    private final double cpuCost;
+    private final double maxMemory;
+    private final double maxMemoryWhenOutputting;
```

Comment:
As I understand this field it should be named `peakMemory`

Reply:
you can call `maxMemory` a `peakMemory`, but then you would call `maxMemoryWhenOutputting` a `peakMemoryWhenOutputting`... which is not "peak" any more.
this is why i renamed the fields to their current form.